addons/l10n_in_sale/models/sale_order.py

```diff
@@ -37,6 +37,8 @@ def _compute_l10n_in_gst_treatment(self):
     @api.depends('company_id')
     def _compute_l10n_in_journal_id(self):
         for order in self:
+            if order._origin and order._origin.l10n_in_company_country_code == 'IN':
```

Comment:
Hello @jbw-odoo ! Thank you for your insight! I implemented your suggestion and also added that the modified journal should belong to the same company as the order, so that the journal will always be changed accordingly when the user do multiple modifications without saving in between. I omitted the original condition where "order._origin.l10n_in_company_country_code == 'IN' " since it would not be of any help if the database has more than one Indian company.